aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -354,23 +410,26 @@ def route(
         cors: Optional[bool] = None,
         compress: bool = False,
         cache_control: Optional[str] = None,
-        middleware: Optional[List[Callable]] = None,
+        middlewares: Optional[List[Callable[..., Any]]] = None,
     ):
         raise NotImplementedError()
 
-    def use(self, middleware: Union[Callable, List[Callable]]):
-        """Add a middleware to the router"""
-        if not isinstance(middleware, list):
-            middleware = [middleware]
-        self.router_middleware = list(self.router_middleware) + list(middleware)
+    def use(self, middlewares: List[Callable[..., Any]]):
+        """
+        Add a List of middlewares to the global routetr middleware list
+
+        These Middleware handlers will be executed in order of adding to the array and will
+        be executed before any specific middleware applied at the actual route level.
```

Comment:
```suggestion
        These middlewares will be called in insertion order and 
        before any middleware registered at the route level.
```

we avoid using `execute` verb like this, since it can be misinterpreted by ESLs.

Reply:
We're missing test for these. Also test for when `Router` is used (middlewares registered in split route scenario)